### PR TITLE
Kenny Lab

### PR DESCRIPTION
**Pull request by KorbKenny**
I am crazy lost with the RecyclerView Adapter. Even reading about it, there are so many things I'm typing in randomly. I commented out where I was super lost. I'd love some extra help on RecyclerViews! (And base adapters, for that matter)

### CODE REVIEW

File: MyApplication/app/src/main/java/com/korbkenny/recyclerview_lab/RecyclerViewAdapter.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    public CustomViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
+        View parentView = LayoutInflater.from(parent.getContext()).inflate(R.layout.view_item, parent, false);
+        CustomViewHolder viewHolder = new CustomViewHolder(parentView);
+        return viewHolder;
+    }
+
+    @Override
+    public void onBindViewHolder(CustomViewHolder holder, int position) {
+
+        View.OnClickListener onClickListener = new View.OnClickListener() {
```
Comment (drew-mahrt) [inline MyApplication/app/src/main/java/com/korbkenny/recyclerview_lab/RecyclerViewAdapter.java:28]: You need to separate the concepts of what you do when clicking on the item from the code used for updating the view itself. What goes in the Click Listener is just code for setting the boolean for the checkbox to either true or false, notifying the adapter that it needs to update, and then show a toast.

Outside of the click listener (after the ending curly brace) you put the code that accesses the view holder to set up each of the views. This includes looking at the current object to determine color, title, status of the checkbox, etc. and using those values to set properties for the views.

Basically code in the click listener updates your data (the objects), code outside of it directly updates your view.

File: MyApplication/app/src/main/res/layout/view_item.xml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+            android:layout_below="@id/TitleText"/>
+        
+    </RelativeLayout>
+
+    <CheckBox
+        android:layout_margin="10dp"
+        android:id="@+id/checkbox"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:onClick="Checkbox has been clicked"/>
```
Comment (drew-mahrt) [inline MyApplication/app/src/main/res/layout/view_item.xml:39]: Don't use onClick like this, it is meant for assigning a method name to be executed when the view is clicked.

**Comment by charlesdrews** [comment]
The `onBindView()` method in the adapter is really the heart of this assignment, and since that is not set up, I'm afraid I'll have to mark this lab as incomplete. The idea is to use the `position` parameter to access the right object from `mObjectList` then use the view holder parameter as a means to access the specific views (text views, checkbox, etc) and assign values from your custom object into those views.
